### ISSUE TITLE
[Bug, a11y]: Search results page: Titles links in search results list are not semantically marked as a heading

### ISSUE THREAD

**Issue by CoffeTheraphy** [coveo/ui-kit#6063]
### Which product are you using?

Atomic

### product version

"@coveo/atomic-react": "3.2.6"

### bug description

Search results page: Titles links in search results list are not semantically marked as a heading

### Steps to reproduce

**User impact:** Without semantic headings for article links in the search results, screen reader users cannot quickly identify the structure or hierarchy of the content. This makes it harder for users to efficiently navigate or understand the importance of each item in the results list, leading to a frustrating and disjointed experience. Proper semantic headings improve accessibility by **providing** meaningful context and aiding faster navigation for users who rely on assistive technologies.

**STEPS TO REPRODUCE:**
1. Open the page https://docs.coveo.com/en/atomic/latest/reference/result-template-components/atomic-result-section-title/
2. Perform any valid search that brings results (e.g. "coveo")
3. Using the Tab key, navigate to any found title link in the search results list.
4. Listen to the screen reader announcement.

**ACTUAL RESULT:** The link is announced as "Clickable <link title> link", e.g. "Services Coveo Platform, link"
Failed Checkpoint: WCAG SC 1.3.1 https://www.w3.org/WAI/WCAG22/Understanding/info-and-relationships

### Relevant log output

```shell
**EXPECTED RESULT:** The link should be announced as heading of the appropriate level, e.g. "<link title> link heading level N".

Ensure that the title link heading is semantically marked as a heading using the appropriate <hX> element, see https://www.w3.org/WAI/WCAG22/Techniques/html/H42
```

**Comment by fbeaudoincoveo** [comment]
Hi @CoffeTheraphy !

Thanks for reporting this. This is a valid accessibility concern.

The `atomic-result-section-title` is entirely optional and is meant to be used for styling purposes in the result template. While it is often used as a wrapper around the `atomic-result-link`, it could be omitted entirely or used to wrap other elements. It does not currently render a semantic `h` element.

However, the `atomic-result-link` exposes an `attributes` slot in its public API that allows you to override attributes on the `a` element it outputs. This is frequently used to specify the desired `target` behavior, but it can also be used to set ARIA attributes.

I would suggest using the `aria-roledescription` attribute to address the accessibility concern:

```html
<atomic-result-section-title>
  <atomic-result-link>
    <a slot="attributes" aria-roledescription="link heading level 2"></a>
  </atomic-result-link>
</atomic-result-section-title>
```

Assistive technology will then announce the element correctly.

Given that there is already a viable workaround for this issue, I will close it.
